lib/core/proxy.c

```diff
@@ -456,8 +457,13 @@ static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errs
         h2o_iovec_t value = headers[i].value;
         if (h2o_iovec_is_token(headers[i].name)) {
             const h2o_token_t *token = H2O_STRUCT_FROM_MEMBER(h2o_token_t, buf, headers[i].name);
-            if (token->flags.proxy_should_drop_for_res)
+            if (token->flags.proxy_should_drop_for_res) {
+                if (forward_close_connection && token == H2O_TOKEN_CONNECTION && self->src_req->version < 0x200) {
```

Comment:
Can we change this to `token == H2O_TOKEN_CONNECTION && self->src_req->version < 0x200 && req->conn->ctx->globalconf->proxy.forward_close_connection`, thereby eliminating `forward_close_connection`?

By doing so, handling of the configuration becomes very localized, as well as eliminating the additional computing cost (when http2client is used).

Reply:
Sure thing. I see your point on optimizing for better reference locality, especially in a code path like this.

Reply:
Done: 490594e02424b56c0d68ffddd1ea576cedabf1ac

Lemme know if you think the line is too long for your taste. I don't have an opinion on this.